### PR TITLE
Add dev variant with xdebug, blackfire, ...

### PR DESCRIPTION
**Pull request by weitzman**
New code at bottom of Dockerfile. All else is same as /ci

### CODE REVIEW

File: dev/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+ENV NVM_DIR /usr/local/nvm
+ENV NODE_VERSION v6.11.0
+ENV NODE_PATH $NVM_DIR/versions/node/$NODE_VERSION/lib/node_modules
+ENV PATH $NVM_DIR/versions/node/$NODE_VERSION/bin:$PATH
+ENV PHANTOMJS_VERSION 2.1.1
+
+# Install development tools.
+RUN pecl install xdebug \
+  && docker-php-ext-enable xdebug \
+  && echo "xdebug.default_enable=0\nxdebug.coverage_enable=0" > $PHP_INI_DIR/conf.d/xdebug.ini \
```
Comment (rbayliss) [inline dev/Dockerfile]: I think we can actually get rid of the following lines:
```
  # Install PhantomJS
  && apt-get update \
  # Install fonts and libtdl (for mounted docker binary usage)
  && apt-get install -y libfontconfig libltdl7 \
  && curl -sL "https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-${PHANTOMJS_VERSION}-linux-x86_64.tar.bz2" | tar jx \
	&& mv phantomjs-${PHANTOMJS_VERSION}-linux-x86_64 /usr/local/bin/phantomjs \
.... Later on:
  # Clean up the leftovers
	&& npm cache clean --force \
  && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
```

File: dev/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+  # Install fonts and libtdl (for mounted docker binary usage)
+  && apt-get install -y libfontconfig libltdl7 \
+  && curl -sL "https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-${PHANTOMJS_VERSION}-linux-x86_64.tar.bz2" | tar jx \
+	&& mv phantomjs-${PHANTOMJS_VERSION}-linux-x86_64 /usr/local/bin/phantomjs \
+	# Install Blackfire Probe
+	&& version=$(php -r "echo PHP_MAJOR_VERSION.PHP_MINOR_VERSION;") \
+  && curl -A "Docker" -o /tmp/blackfire-probe.tar.gz -D - -L -s https://blackfire.io/api/v1/releases/probe/php/linux/amd64/$version \
+  && tar zxpf /tmp/blackfire-probe.tar.gz -C /tmp \
+  && mv /tmp/blackfire-*.so $(php -r "echo ini_get('extension_dir');")/blackfire.so \
+  && printf "extension=blackfire.so\nblackfire.agent_socket=tcp://blackfire:8707\n" > $PHP_INI_DIR/conf.d/blackfire.ini \
```
Comment (rbayliss) [inline dev/Dockerfile]: This configuration requires a `blackfire` container running alongside drupal, `blackfire/blackfire` - See https://github.com/LastCallMedia/Drupal-Scaffold/blob/master/docker-compose.yml#L74.  Nothing to change here, but if we want "out of the box" blackfire (I certainly do), we'll need to add it to our `docker-compose.yml`

File: dev/etc/php-overrides.ini [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+; the container at $PHP_INI_DIR/conf.d/zz-overrides.ini
+
+upload_max_filesize = 128M
+post_max_size = 128M
+memory_limit = 256M
+sendmail_path = /bin/true
+date.timezone = UTC
+
+;This is for APCu.
+apc.ttl=7200
```
Comment (rbayliss) [inline dev/etc/php-overrides.ini]: APCu++

File: dev/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  && apt-get install -y apt-transport-https lsb-release > /dev/null 2>&1 \
+  && curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add - \
+  && echo 'deb https://deb.nodesource.com/node_6.x jessie main' > /etc/apt/sources.list.d/nodesource.list \
+  && echo 'deb-src https://deb.nodesource.com/node_6.x jessie main' >> /etc/apt/sources.list.d/nodesource.list \
+  && apt-get update \
+  && apt-get install nodejs \
+  && npm install -g yarn \
+  && npm cache clean --force \
+  && apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
```
Comment (rbayliss) [inline dev/Dockerfile]: I think we can get rid of the whole section that follows this, which is:
```
# Install development tools.
# We install nvm, but also add node to $PATH directly
# so we can use it through /bin/sh.
ENV NVM_DIR /usr/local/nvm
ENV NODE_VERSION v6.11.0
ENV NODE_PATH $NVM_DIR/versions/node/$NODE_VERSION/lib/node_modules
ENV PATH $NVM_DIR/versions/node/$NODE_VERSION/bin:$PATH
ENV PHANTOMJS_VERSION 2.1.1
```

That stuff is needed for LCM's installation of NVM and PhantomJS, but we don't need it here.

File: dev/Dockerfile (hunk 1 above)
Comment (yriahi) [reply dev/Dockerfile]: I think it may not be a bad idea to keep if we'd like to extend testing automation. We can also ad later.

Comment (rbayliss) [reply dev/Dockerfile]: I don't think PhantomJS is the direction I'd recommend for front end testing these days.  There are much better alternatives with http://webdriver.io/ and Puppeteer for driving headless Chrome. This is a legacy thing from Last Call's Docker boxes.

Comment (yriahi) [reply dev/Dockerfile]: gotcha 👍

**Pull request merged**